evalml/pipelines/components/transformers/imputers/knn_imputer.py

```diff
@@ -0,0 +1,158 @@
+            X = X.ww.copy()
+            X.ww.set_types({col: "Categorical" for col in boolean_columns})
+        return X
+
+    def fit(self, X, y=None):
+        """Fits imputer to data. 'None' values are converted to np.nan before imputation and are treated as the same.
+
+        Args:
+            X (pd.DataFrame or np.ndarray): the input training data of shape [n_samples, n_features]
+            y (pd.Series, optional): the target training data of length [n_samples]
+
+        Returns:
+            self
+
+        Raises:
+            ValueError: if the KNNImputer receives a dataframe with both Boolean and Categorical data.
+
+        """
+        X = infer_feature_types(X)
+
+        if set([lt.type_string for lt in X.ww.logical_types.values()]) == {
+            "boolean",
+            "categorical",
+        } and not all(
+            [
+                is_categorical_actually_boolean(X, col)
+                for col in X.ww.select("Categorical")
+            ],
+        ):
+            raise ValueError(
+                "KNNImputer cannot handle dataframes with both boolean and categorical features.  Use Imputer instead.",
+            )
+
+        nan_ratio = X.ww.describe().loc["nan_count"] / X.shape[0]
+        self._all_null_cols = nan_ratio[nan_ratio == 1].index.tolist()
+
+        X, _ = self._drop_natural_language_columns(X)
+        X = self._set_boolean_columns_to_categorical(X)
+
+        # If the Dataframe only had natural language columns, do nothing.
+        if X.shape[1] == 0:
+            return self
+
+        self._component_obj.fit(X, y)
+        return self
+
+    def transform(self, X, y=None):
+        """Transforms input by imputing missing values. 'None' and np.nan values are treated as the same.
+
+        Args:
+            X (pd.DataFrame): Data to transform.
+            y (pd.Series, optional): Ignored.
+
+        Returns:
+            pd.DataFrame: Transformed X
+        """
+        X = infer_feature_types(X)
+
+        # Return early since bool dtype doesn't support nans and sklearn errors if all cols are bool
+        if (X.dtypes == bool).all():
+            return X
+
+        not_all_null_cols = [col for col in X.columns if col not in self._all_null_cols]
+        original_index = X.index
+
+        # Drop natural language columns and transform the other columns
+        X_t, natural_language_cols = self._drop_natural_language_columns(X)
+        if X_t.shape[-1] == 0:
+            return X
+        not_all_null_or_natural_language_cols = [
+            col for col in not_all_null_cols if col not in natural_language_cols
+        ]
+        X_t = self._component_obj.transform(X_t)
+        X_t = pd.DataFrame(X_t, columns=not_all_null_or_natural_language_cols)
+
+        # Add back in natural language columns, unchanged
+        if len(natural_language_cols) > 0:
+            X_t = woodwork.concat_columns([X_t, X[natural_language_cols]])
+
+        booleanNullable_columns = list(
```

Comment:
if we're imputing these `BooleanNullable` cols we can cast them to regular `Boolean` columns since they won't have null values anymore. I have a [PR](https://github.com/alteryx/evalml/pull/3678/files#diff-6daa63a0b019305d4ef1d44d1be7ac73769d1912ff22486bd551854d723d4393R176) up that does it for `Imputer.py`